converter/internal/common/common.go

```diff
@@ -32,3 +39,38 @@ func getValueOverrideHook() builder.ValueOverrideHook {
 		}
 	}
 }
+
+func GetUniqueLabel(label string, currentCount int) string {
+	if currentCount == 1 {
+		return label
+	}
+
+	return fmt.Sprintf("%s_%d", label, currentCount)
+}
+
+// prettyPrint attempts to pretty print the input slice. If prettyPrint fails,
+// the input slice is returned unmodified.
+func PrettyPrint(in []byte) ([]byte, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	// Return early if there was no file.
+	if len(in) == 0 {
+		return in, diags
+	}
+
+	f, err := parser.ParseFile("", in)
+	if err != nil {
+		diags.Add(diag.SeverityLevelWarn, err.Error())
+		return in, diags
+	}
+
+	var buf bytes.Buffer
+	if err := printer.Fprint(&buf, f); err != nil {
+		diags.Add(diag.SeverityLevelWarn, err.Error())
+		return in, diags
+	}
+
+	// Add a trailing newline at the end of the file, which is omitted by Fprint.
+	_, _ = buf.Write([]byte{'\n'})
```

Comment:
WriteString may save a conversion though I suspect its the exact same performance, just a bit clearer.